### ISSUE TITLE
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'

### ISSUE THREAD

**Issue by Kennygiflicker** [FujiMakoto/pygelbooru#12]
Hello, so i just code a discord bot but i just get this error from your pygelbooru do you know what happend?

Here is my code:
```
@bot.command()
async def random(ctx, arg: str):
    pic = await gelbooru.random_post(tags=[arg])
    await ctx.send(pic)
```
Here is error:
```
botgelbooru is ready
Ignoring exception in command random:
Traceback (most recent call last):
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "<string>", line 16, in random
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 216, in random_post
    return await self.search_posts(tags=tags, exclude_tags=exclude_tags, limit=1, page=offset)
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 257, in search_posts
    else [GelbooruImage(payload['posts']['post'], self)]
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 32, in __init__
    self.id             = int(payload.get('@id'))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/bot.py", line 939, in invoke
    await ctx.command.invoke(ctx)
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 863, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 94, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```
Sorry for my bad english

**Comment by Etwodev** [comment]
> Hello, so i just code a discord bot but i just get this error from your pygelbooru do you know what happend?
> 
> Here is my code:
> 
> ```
> @bot.command()
> async def random(ctx, arg: str):
>     pic = await gelbooru.random_post(tags=[arg])
>     await ctx.send(pic)
> ```
> 
> Here is error:
> 
> ```
> botgelbooru is ready
> Ignoring exception in command random:
> Traceback (most recent call last):
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 85, in wrapped
>     ret = await coro(*args, **kwargs)
>   File "<string>", line 16, in random
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 216, in random_post
>     return await self.search_posts(tags=tags, exclude_tags=exclude_tags, limit=1, page=offset)
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 257, in search_posts
>     else [GelbooruImage(payload['posts']['post'], self)]
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 32, in __init__
>     self.id             = int(payload.get('@id'))
> TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/bot.py", line 939, in invoke
>     await ctx.command.invoke(ctx)
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 863, in invoke
>     await injected(*ctx.args, **ctx.kwargs)
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 94, in wrapped
>     raise CommandInvokeError(exc) from exc
> discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
> ```
> 
> Sorry for my bad english

Do the non-async methods work at all too? I'm also having this issue.

**Comment by Etwodev** [comment]
> Hello, so i just code a discord bot but i just get this error from your pygelbooru do you know what happend?
> 
> Here is my code:
> 
> ```
> @bot.command()
> async def random(ctx, arg: str):
>     pic = await gelbooru.random_post(tags=[arg])
>     await ctx.send(pic)
> ```
> 
> Here is error:
> 
> ```
> botgelbooru is ready
> Ignoring exception in command random:
> Traceback (most recent call last):
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 85, in wrapped
>     ret = await coro(*args, **kwargs)
>   File "<string>", line 16, in random
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 216, in random_post
>     return await self.search_posts(tags=tags, exclude_tags=exclude_tags, limit=1, page=offset)
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 257, in search_posts
>     else [GelbooruImage(payload['posts']['post'], self)]
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/pygelbooru/gelbooru.py", line 32, in __init__
>     self.id             = int(payload.get('@id'))
> TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/bot.py", line 939, in invoke
>     await ctx.command.invoke(ctx)
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 863, in invoke
>     await injected(*ctx.args, **ctx.kwargs)
>   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/discord/ext/commands/core.py", line 94, in wrapped
>     raise CommandInvokeError(exc) from exc
> discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
> ```
> 
> Sorry for my bad english

Seems as if maybe the API has changed?
```
OrderedDict([('posts', OrderedDict([('@limit', '100'), ('@offset', '0'), ('@count', '1'), ('post', OrderedDict([('id', '6186618'), ('created_at', 'Sun Jun 13 03:45:25 -0500 2021'), ('score', '1'), ('width', '1536'), ('height', '2048'), ('md5', 'cef1d3d2668f8f8c01e792ca42494e1d'), ('directory', 'ce/f1'), ('image', 'cef1d3d2668f8f8c01e792ca42494e1d.jpeg'), ('rating', 'safe'), ('source', 'https://mobile.twitter.com/kiku_pond/status/1403891174968991749'), ('change', '1625171038'), ('owner', 'raptorq64'), ('creator_id', '198849'), ('parent_id', '0'), ('sample', '1'), ('preview_height', '250'), ('preview_width', '187'), ('tags', '1girl :o artist_name artist_request birthday blonde_hair blush braid breasts buttons collarbone crown_braid female_focus green_eyes green_neckwear grey_sailor_collar hair_rings happy_birthday highres light_brown_eyes long_sleeves looking_at_viewer love_live! love_live!_sunshine!! neckerchief ohara_mari parted_lips sailor_collar school_uniform shirt short_hair smile solo surprised tie_clip upper_body uranohoshi_school_uniform white_shirt wow yellow_buttons yellow_eyes'), ('title', None), ('has_notes', 'false'), ('has_comments', 'false'), ('file_url', 'https://img3.gelbooru.com/images/ce/f1/cef1d3d2668f8f8c01e792ca42494e1d.jpeg'), ('preview_url', 'https://img3.gelbooru.com/thumbnails/ce/f1/thumbnail_cef1d3d2668f8f8c01e792ca42494e1d.jpg'), ('sample_url', 'https://img3.gelbooru.com/samples/ce/f1/sample_cef1d3d2668f8f8c01e792ca42494e1d.jpg'), ('sample_height', '1133'), ('sample_width', '850'), ('status', 'active'), ('post_locked', '0'), ('has_children', 'false')]))]))])
```
Fixed by removing the @'s before every get operation

```py
        self.id = int(payload.get('@id'))
        self.name = payload.get('@name')
        self.count = int(payload.get('@count', 0))
        self.ambiguous = payload.get('@ambiguous')
```
becomes 
```py
        self.id = int(payload.get('id'))
        self.name = payload.get('name')
        self.count = int(payload.get('count', 0))
        self.ambiguous = payload.get('ambiguous')
```

**Comment by FujiMakoto** [comment]
Resolved in #13
